editor/plugins/animation_state_machine_editor.cpp

```diff
@@ -1601,6 +1623,205 @@ void AnimationNodeStateMachineEditor::_update_mode() {
 	}
 }
 
+void AnimationNodeStateMachineEditor::_node_group_edit_redraw(const StringName &p_node_name) {
+	if (side_panel_mode == NODE_GROUP_EDIT) {
+		if (p_node_name) {
+			sidepanel_title_2->set_text(String(p_node_name));
+			sidepanel_title_2->add_theme_color_override(StringName("font_color"), Color(.55, .93, .59));
+			for (int i = 0; i < 12; i++) {
+				node_group_select_checkboxes[i]->set_visible(true);
+				node_group_select_checkboxes[i]->set_pressed(bool(state_machine->_get_node_group_selections(p_node_name)[i]));
+			}
+		} else {
+			sidepanel_title_2->set_text(TTR("Select An Animation Node"));
+			sidepanel_title_2->add_theme_color_override(StringName("font_color"), Color(1, .5, .5));
+			for (int i = 0; i < 12; i++) {
+				node_group_select_checkboxes[i]->set_visible(false);
+			}
+		}
+	}
+}
+
+void AnimationNodeStateMachineEditor::_draw_sidepanel_edit_node_group() {
+	side_panel_mode = NODE_GROUP_EDIT;
+
+	sidepanel_title->set_text(TTR("Animation Node:"));
+	return_to_node_group_selection_menu_button->set_text(TTR("Return"));
+	return_to_node_group_selection_menu_button->set_visible(true);
+	sidepanel_title_2->set_visible(true);
+	show_all_node_group_button->set_visible(false);
+	hide_all_node_group_button->set_visible(false);
+	node_group_selection_menu_button->set_visible(false);
+	node_group_select_menu_button->set_visible(false);
+	node_group_name_edit_menu_button->set_visible(false);
+	node_group_selection_menu_button->set_visible(false);
+
+	if (selected_node) {
+		sidepanel_title_2->set_text(String(selected_node));
+		sidepanel_title_2->add_theme_color_override(StringName("font_color"), Color(.55, .93, .59));
+		for (int i = 0; i < 12; i++) {
+			node_group_select_checkboxes[i]->set_visible(true);
+			node_group_select_checkboxes[i]->set_pressed(bool(state_machine->_get_node_group_selections(selected_node)[i]));
+		}
+
+	} else {
+		sidepanel_title_2->set_text(TTR("Select An Animation Node"));
+		sidepanel_title_2->add_theme_color_override(StringName("font_color"), Color(1, .5, .5));
+	}
+
+	for (int i = 0; i < 12; i++) {
+		show_node_group_buttons[i]->set_visible(false);
+		hide_node_group_buttons[i]->set_visible(false);
+		node_group_labels[i]->set_visible(true);
+		node_group_labels[i]->set_text(state_machine->_get_node_group_names()[i]);
+		node_group_name_line[i]->set_visible(false);
+	}
+}
+
+void AnimationNodeStateMachineEditor::_draw_sidepanel_node_group() {
+	PackedStringArray node_group_names = state_machine->_get_node_group_names();
+
+	// !updating is for keeping the name consistent when changing between animation trees while entering a new node group name
+	if ((side_panel_mode == NODE_GROUP_NAME_EDIT) & (!updating)) {
```

Comment:
```suggestion
	if ((side_panel_mode == NODE_GROUP_NAME_EDIT) && !updating) {
```
Please use `&&` when you're using Boolean values